cmake/Config.cmake

```diff
@@ -17,6 +17,11 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
         message(FATAL_ERROR "Unsupported architecture")
         return()
     endif()
+
+    #compile for ARM64 if set
+    if(${CMAKE_GENERATOR_PLATFORM} MATCHES "ARM64")
+        set(ARCH_ARM64 1)
+    endif()
```

Comment:
I think we need to change some variable names to make it clear we're talking about 32 and 64 bit Intel architectures. Also I think all 3 of these architectures, which are independent, should be checked in the same `if`/`else if` chain because we don't want `ARCH_64BITS` and `ARCH_ARM64` set at the same time which is something that is currently happening.

This is what I propose.

```cmake
    if(${CMAKE_GENERATOR_PLATFORM} MATCHES "ARM64")
        set(ARCH_ARM64 1)
    elseif(${SIZEOF_VOID_PTR} STREQUAL "4")
        set(ARCH_X86 1)
    elseif(${SIZEOF_VOID_PTR} STREQUAL "8")
        set(ARCH_X64 1)
    else()
        message(FATAL_ERROR "Unsupported architecture")
        return()
    endif()
```